server/apps/microservices/src/processors/metadata-extraction.processor.ts

```diff
@@ -345,3 +371,85 @@ export class MetadataExtractionProcessor {
     return Duration.fromObject({ seconds: videoDurationInSecond }).toFormat('hh:mm:ss.SSS');
   }
 }
+
+@Processor(QueueName.SIDECAR)
+export class SidecarProcessor {
+  private logger = new Logger(SidecarProcessor.name);
+  private assetCore: AssetCore;
+
+  constructor(
+    @Inject(IAssetRepository) private assetRepository: IAssetRepository,
+    @Inject(IJobRepository) private jobRepository: IJobRepository,
+  ) {
+    this.assetCore = new AssetCore(assetRepository, jobRepository);
+  }
+
+  @Process(JobName.QUEUE_SIDECAR)
+  async handleQueueSidecar(job: Job<IBaseJob>) {
+    try {
+      const { force } = job.data;
+      const assetPagination = usePagination(JOBS_ASSET_PAGINATION_SIZE, (pagination) => {
+        return force
+          ? this.assetRepository.getWith(pagination, WithProperty.SIDECAR)
+          : this.assetRepository.getWithout(pagination, WithoutProperty.SIDECAR);
+      });
+
+      for await (const assets of assetPagination) {
+        for (const asset of assets) {
+          const name = force ? JobName.SIDECAR_SYNC : JobName.SIDECAR_DISCOVERY;
+          await this.jobRepository.queue({ name, data: { asset } });
+        }
+      }
+    } catch (error: any) {
+      this.logger.error(`Unable to queue sidecar scanning`, error?.stack);
+    }
+  }
+
+  @Process(JobName.SIDECAR_SYNC)
+  async handleSidecarSync(job: Job<IAssetJob>) {
+    const { asset } = job.data;
+    if (!asset.isVisible) {
+      return;
+    }
+
+    try {
+      const fileName = asset.originalFileName;
+      const name = asset.type === AssetType.VIDEO ? JobName.EXTRACT_VIDEO_METADATA : JobName.EXIF_EXTRACTION;
+      await this.jobRepository.queue({ name, data: { asset, fileName } });
+    } catch (error: any) {
+      this.logger.error(`Unable to queue metadata extraction`, error?.stack);
+    }
+  }
+
+  @Process(JobName.SIDECAR_DISCOVERY)
+  async handleSidecarDiscovery(job: Job<IAssetJob>) {
+    let { asset } = job.data;
+    if (!asset.isVisible) {
+      return;
+    }
+
+    if (asset.sidecarPath) {
+      return;
+    }
+
+    try {
+      await fs.promises.access(`${asset.originalPath}.xmp`, fs.constants.W_OK);
+
+      try {
+        asset = await this.assetCore.save({ id: asset.id, sidecarPath: `${asset.originalPath}.xmp` });
+        const fileName = asset.originalFileName;
+        // TODO: optimize to only queue assets with recent xmp changes
+        const name = asset.type === AssetType.VIDEO ? JobName.EXTRACT_VIDEO_METADATA : JobName.EXIF_EXTRACTION;
+        await this.jobRepository.queue({ name, data: { asset, fileName } });
```

Comment:
`fileName` isn't included in this job anymore.

Reply:
Cleaned up.